src/libos/src/poll/do_epoll/epoll_file.rs

```diff
@@ -290,6 +291,20 @@ impl EpollFile {
     }
 }
 
+impl Drop for EpollFile {
+    fn drop(&mut self) {
+        let mut interest = self.interest.lock().unwrap();
+        for (fd, entry) in interest.drain() {
+            entry.set_deleted();
+            if let Some(file) = entry.file() {
+                file.unregister_observer(&(entry as _));
+            }
+
+            self.unregister_from_file(fd);
+        }
```

Comment:
```suggestion
    fn drop(&mut self) {
        let mut interest = self.interest.lock().unwrap();
        let fds: Vec<_> = interest
            .drain()
            .map(|(fd, entry)| {
                entry.set_deleted();
                if let Some(file) = entry.file() {
                    file.unregister_observer(&(entry as _));
                }
                fd
            })
            .collect();
        drop(interest);

        fds.iter().for_each(|fd| self.unregister_from_file(*fd));
    }
```

Reply:
Reference EpollCtl::Del implemetation, we should hold the  interest lock and file lock separately.